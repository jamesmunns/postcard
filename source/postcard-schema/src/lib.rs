#![cfg_attr(not(any(test, feature = "use-std")), no_std)]
#![warn(missing_docs)]
#![cfg_attr(docsrs, feature(doc_cfg))]
//! # Postcard Schema

pub mod impls;
pub mod schema;

#[cfg(feature = "derive")]
pub use postcard_derive::Schema;

/// A trait that represents a compile time calculated schema
pub trait Schema {
    /// A recursive data structure that describes the schema of the given
    /// type.
    const SCHEMA: &'static schema::NamedType;
}
