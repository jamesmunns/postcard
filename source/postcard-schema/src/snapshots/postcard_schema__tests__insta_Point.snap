---
source: source/postcard-schema/src/lib.rs
expression: "Point :: SCHEMA"
---
NamedType {
    name: "Point",
    ty: Struct(
        [
            NamedValue {
                name: "x",
                ty: NamedType {
                    name: "i32",
                    ty: I32,
                },
            },
            NamedValue {
                name: "y",
                ty: NamedType {
                    name: "i32",
                    ty: I32,
                },
            },
        ],
    ),
}
